pkg/controller/appstatus/status.go

```diff
@@ -481,7 +481,11 @@ func AppStatus(req router.Request, resp router.Response) error {
 
 		status := container[containerName]
 		status.Ready = dep.Status.ReadyReplicas
-		status.ReadyDesired = dep.Status.Replicas
+		if dep.Spec.Replicas != nil {
```

Comment:
Hmm this should be dep.status.Replica?

Reply:
No, the `status.Replicas` has the possibility to be 0 (not set) if something goes wrong. `spec.Replicas` is a pointer so I have to do the nil check first before proceeding.

Reply:
But dep.spec.replicas will never be the replica that is ready? It is the desired replica for a deployment, not the ready replica.

Reply:
oh wait, this is `ReadyDesired`.... yeah setting spec.replica makes sense...